state/machine.go

```diff
@@ -1065,6 +1065,70 @@ func (m *Machine) Units() (units []*Unit, err error) {
 	return units, nil
 }
 
+// XXX(jam): 2016-12-09 These are just copied from
+// provider/maas/constraints.go, but they should be tied to machine
+// constraints, *not* tied to provider/maas constraints.
+// convertSpacesFromConstraints extracts spaces from constraints and converts
+// them to two lists of positive and negative spaces.
+func convertSpacesFromConstraints(spaces *[]string) ([]string, []string) {
+	if spaces == nil || len(*spaces) == 0 {
+		return nil, nil
+	}
+	return parseDelimitedValues(*spaces)
+}
+
+// parseDelimitedValues parses a slice of raw values coming from constraints
+// (Tags or Spaces). The result is split into two slices - positives and
+// negatives (prefixed with "^"). Empty values are ignored.
+func parseDelimitedValues(rawValues []string) (positives, negatives []string) {
```

Comment:
I just copied these 2 functions from the provider/maas directory, because it *really* doesn't belong there. It either belongs inside something to do with "constraints" (probably my preference), or something to do with "networking". Thoughts?